dyno-core/src/main/java/com/netflix/dyno/connectionpool/Host.java

```diff
@@ -152,6 +174,12 @@ public boolean equals(Object obj) {
             return false;
 
         Host other = (Host) obj;
+        /* we need a way to pass the information about the hashtag from the token map supplier
+         * to the host object. 
+         */
+        if (other.hashtag != null) {
```

Comment:
why are we setting the hashtag here? Shouldnt equals be readonly?

Reply:
There are two pieces of information that we need to combine to determine healthy nodes.The host supplier (Eureka) and the host pool (provided by DM). The host supplier does not have information about the hashtags. Hence, in the simple operations, we use the host pool to derive the hashtags. Then we pass the hashtags to the `getPoolForOperation` such that we can identify the node based on the hashed value.

However, the Pipeline operations have been developed in a different way. A pipeline selects directly a Host object. The Host object is created based on the Host supplier, which does not have the hashtags (Eureka does not provide hashtags). I could not find a way to let the host pool inject it in the Host object since the equality is the only call happening to combine these two.

Reply:
I am only saying making changes while calling "compareTo" or Equals is not good.
Those functions should not have any sideeffects

Reply:
You are right, let me think about it. I have committed all the rest of the fixes.